run/service-auth/receive_test.py

```diff
@@ -101,19 +124,75 @@ def services():
     )
 
 
-def test_auth(services):
-    url = services[0]
-    token = services[1]
+def test_authentication_on_cloud_run(service: tuple[str, str]) -> None:
+    endpoint_url = service[0]
+    token = service[1]
+
+    req = request.Request(endpoint_url)
+    try:
+        _ = request.urlopen(req)
+    except error.HTTPError as e:
+        assert e.code == HTTP_STATUS_FORBIDDEN
+
+    retry_strategy = Retry(
+        total=3,
+        status_forcelist=STATUS_FORCELIST,
+        allowed_methods=["GET", "POST"],
+        backoff_factor=3,
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+
+    client = requests.session()
+    client.mount("https://", adapter)
+
+    response = client.get(endpoint_url, headers={"Authorization": f"Bearer {token}"})
+    response_content = response.content.decode("UTF-8")
+
+    assert response.status_code == HTTP_STATUS_OK
+    assert "Hello" in response_content
+    assert "anonymous" not in response_content
+
+
+def test_anonymous_request_on_cloud_run(service: tuple[str, str]) -> None:
+    endpoint_url = service[0]
+
+    req = request.Request(endpoint_url)
+    try:
+        _ = request.urlopen(req)
+    except error.HTTPError as e:
+        assert e.code == HTTP_STATUS_FORBIDDEN
+
+    retry_strategy = Retry(
+        total=3,
+        status_forcelist=STATUS_FORCELIST,
```

Comment:
From earlier in the comments: this is the only time you're using most of the http status codes.

Instead of re-creating your own magic variables, you can use the existing lists from either the core standard library [`HTTPStatus`](https://docs.python.org/3/library/http.html#http.HTTPStatus), or from [`requests.codes`](https://requests.readthedocs.io/en/latest/api/#requests.codes), since that'a already imported.